drivers/hmis/app/graphql/mutations/create_client_alert.rb

```diff
@@ -0,0 +1,20 @@
+module Mutations
+  class CreateClientAlert < CleanBaseMutation
+    argument :input, Types::HmisSchema::ClientAlertInput, required: true
+
+    field :client_alert, Types::HmisSchema::ClientAlert, null: true
+    field :errors, [Types::HmisSchema::ValidationError], null: false, resolver: Resolvers::ValidationErrors
+
+    def resolve(input:)
+      raise 'not allowed' unless current_user.can_manage_client_alerts
```

Comment:
I borrowed this line from other examples in the codebase, but I was wondering, on the backend do we need to do more granular permission management -- where we make sure the user has permission to manage client alerts on this particular client? And if so, is there an example elsewhere in the codebase where we're doing that?

Reply:
This is an unexpected state, so that's why we raise an exception. The frontend should never let the user attempt to create an alert if the user lacks permission to do so. It would be nice if we had some kind of error wrapper around authorization but we just haven't added that.

Look for current_permission there are examples for checking permissions against an "entity":
`current_permission?(permission: : can_manage_client_alerts, entity: client)`